File: Examples/test-suite/ruby/function_typedef_runme.rb
#!/usr/bin/env ruby
#
# Tests for function_typedef.i
#
#

require 'swig_assert'
require 'function_typedef'
include Function_typedef

addfunc = Function_typedef.addfunc
swig_assert_equal(do_binop1(addfunc, 11, 22), 33, 'do_binop1')
swig_assert_equal(do_binop2(addfunc, 17, 51), 68, 'do_binop2')
swig_assert_equal(do_binop3(addfunc, 32, 47), 79, 'do_binop3')

File: Examples/test-suite/lua/function_typedef_runme.lua
local v=require("function_typedef")

local addfunc = v.addfunc
assert(swig_type(addfunc) == "binop_t *|int (*)(int,int)")
assert(v.do_binop1(addfunc, 11, 22) == 33)
assert(v.do_binop2(addfunc, 17, 51) == 68)
assert(v.do_binop3(addfunc, 32, 47) == 79)
